### DISCUSSION TITLE
Listener and reactivity

### DISCUSSION THREAD

**Discussion by 0bon** [solidjs/solid#1373] [Q&A]
Suppose I have a setup as follows:

```
const [val, setval] = createSignal();
function hChange()  {
   // do something
}
function Main() {
   return (
      <input type="text name="abc" onChange={hChange} value={val()} />
   );
}
```

Lets assume ```setVal``` is called by a resource that may take a second to load. The element is rendered, then the resource calls ```setVal``` and this updates the ```input``` value. Since the new value is reactive and no keyboard or mouse input was made, the ```onChange``` handler is not triggered. Is there any way  I can trigger the ```onChange``` listener in this instance ?

**Comment by Brendan-csel** [comment]
The browser only raises those events like `onInput` and `onChange` when the user makes changes.

There is nothing stopping you using `createEffect` to do something else when `val()` changes if you need to.

**Comment by 0bon** [comment]
Sure, but I need access to the event object.

**Comment by mdynnl** [comment]
What brendan meant is probably along the lines of `createEffect(() => { doSomething(val()) })` and with this I'm not sure if the event object is even necessary. But surely you can do unnecessary `input.dispatchEvent(new Event('change'))` if you still want and you need an effect for this too.
